sympy/core/relational.py

```diff
@@ -281,20 +281,40 @@ def equals(self, other, failing_expression=False):
 
     def _eval_simplify(self, ratio, measure, rational, inverse):
         r = self
-        r = r.func(*[i.simplify(ratio=ratio, measure=measure,
-                                rational=rational, inverse=inverse)
-                     for i in r.args])
+
+        # XXX Some sympy classes like Set does not have 'simplify' attribute.
+        def try_simp(i):
```

Comment:
Can we add the simplify method to sets if that is what is needed? It seems wrong to add this here